DD_Plug.md

```diff
@@ -0,0 +1,153 @@
+[Plug](https://github.com/elixir-lang/plug) is a specification for composable modules in between web applications and an abstraction layer for connection adapters of different web server. The basic idea of Plug is to unify the concept of a "connection" that you operate on. This differs from other HTTP middleware layers such as Rack, where the request and response are sepearated in the middleware stack.
+
```

Comment:
What do you think about leading with the Plug and Phoenix paragraph (minus the header) and following up with what is now the lead? In exaggerated simplicity:

Plug is at the heart of Phoenix - Endpoints, Controllers, and Routers. . . . 

Plug is a specification . . . 

This gives readers a reason to care up front - everything HTTP related in Phoenix is a plug.